src/porepy/models/constitutive_laws.py

```diff
@@ -3665,23 +3689,12 @@ def thermal_stress(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
                 raise ValueError("Subdomains must be of dimension nd - 1.")
 
         discr = pp.ad.BiotAd(self.stress_keyword, subdomains)
-        alpha = self.biot_coefficient(subdomains)
-        beta = self.solid_thermal_expansion(subdomains)
-        k = self.bulk_modulus(subdomains)
-
-        # Check that both are scalar. Else, the scaling may not be correct.
-        assert isinstance(alpha, pp.ad.Scalar)
-        assert isinstance(beta, pp.ad.Scalar)
-        # The thermal stress should be multiplied by beta and k. Divide by alpha to
-        # cancel that factor from the discretization matrix.
-        stress: pp.ad.Operator = (
-            beta
-            * k
-            / alpha
-            * (
-                discr.grad_p @ self.temperature(subdomains)
-                - discr.grad_p @ self.reference_temperature(subdomains)
-            )
+        stress: pp.ad.Operator = discr.scalar_gradient(
+            self.temperature_variable
```

Comment:
Although in a way I like passing self.temperature_variable here, I think a more consistent thing would be to link it to the discretization keyword ("mechanics"). That's the identifier for discretizations and their parameters and matrices.

Reply:
The discretization keyword is common for all coupling terms, since they are all triggered by the `mechanics` equation. So the identifier of the specific coupling term must either be connected to the variable with which it is associated (e.g. temperature), or with the equation with which the coupling term is associated (`energy_equation`  or similar). Of these options, the variable is to me the natural one.

Reply:
No, I meant the "other" discretization keyword, e.g. "flow".

Reply:
I agree, using 'physics keywords' would be preferable. For the pressure coupling term, the natural key is accessed as `self.darcy_keyword`. For the thermal coupling, there is no single keyword associated with energy balance, but we do have both `self.enthalpy_keyword` and `self.fourier_keyword`. I am inclined to use `enthalpy_keyword` for now, and then possibly revisit the whole keyword concept at a later stage. Does this make sense?